### ISSUE TITLE
Deprecated apiVersion on generated Ingress resources

### ISSUE THREAD

**Issue by aloise** [garden-io/garden#2384]
## Bug

I'm unable to start any services with ingress specified. Garden will receive the trying to run deploy the project in minikube. The same project was working on older garden versions. 

### Current Behavior
`garden dev`

```
Command "/home/aloise/.garden/tools/kubectl/11ac0bbd21058c50/kubectl --context=minikube --namespace=easyrentals-default apply --prune --selector service=gateway --output=json -f -" failed with code 1:

Warning: extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling webhook "validate.nginx.ingress.kubernetes.io": an error on the server ("") has prevented the request from succeeding

Warning: extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
Error from server (InternalError): error when creating "STDIN": Internal error occurred: failed calling webhook "validate.nginx.ingress.kubernetes.io": an error on the server ("") has prevented the request from succeeding
```

### Expected behavior

It was running on older garden + k8s versions.

### Reproducible example

Any project which has ingress is failing.

### Your environment

* **OS:** Arch Linux, 5.12.1
* minikube v1.19.1 + k8s v1.20.2
* garden v0.12.21

**Comment by aloise** [comment]
Might be related (minikube bug):
https://github.com/ansible/awx-operator/issues/205

**Comment by eysi09** [comment]
Ok happy to hear that. But the point stands that we should update the ingress api version on our end. It should be a quick fix.

**Comment by Cpcrook** [comment]
With the latest (4.3.0) release of Docker Desktop on 2021/12/02 (which includes Kubernetes 1.22.x) this issue appears to be rearing it's head.

**Comment by eysi09** [comment]
Hi @Cpcrook. Thanks for reporting this. 

We thought we had already fixed this, so to begin with what version of Garden are you running?

And what type of module are you using (i.e. `container`, `kubernetes` or `helm`). And any other context that you think might be useful would be greatly appreciated.

**Comment by thsig** [comment]
This was fixed in #2907.
